chia/wallet/wallet_node.py

```diff
@@ -865,9 +865,9 @@ async def add_states_from_peer(
         target_concurrent_tasks: int = 30
 
         # Ensure the list is sorted
-
-        before = len(items_input)
-        items = await self.wallet_state_manager.filter_spam(sort_coin_states(items_input))
+        unique_items = set(items_input)
+        before = len(unique_items)
+        items = await self.wallet_state_manager.filter_spam(sort_coin_states(unique_items))
```

Comment:
this is an unrelated comment, but it seems like we should probably filter *before* we sort. There's no point in sorting items that we'll drop later.